stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/ui/robot/compose/ComposeChannelsRobot.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.uitests.ui.robot.compose
+
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import androidx.compose.ui.test.onAllNodesWithContentDescription
+import androidx.compose.ui.test.performClick
+import org.junit.rules.TestRule
+
+/**
+ * A factory function for [ComposeChannelsRobot].
+ */
+internal fun composeChannelsRobot(
+    composeTestRule: ComposeTestRule,
+    block: ComposeChannelsRobot.() -> Unit,
+): ComposeChannelsRobot {
+    return ComposeChannelsRobot(composeTestRule).apply { block() }
+}
+
+/**
+ * A robot that simulates user behavior on the channels screen.
+ *
+ * @param composeTestRule A [TestRule] that provides the main entry point into testing.
+ */
+internal class ComposeChannelsRobot(
```

Comment:
Eventually, we will extract robot interfaces and have 2 implementations for Compose and UI Components as 99% of the methods should be the same.